src/rgw/rgw_sal_motr.cc

```diff
@@ -3469,6 +3677,17 @@ int MotrStore::set_buckets_enabled(const DoutPrefixProvider *dpp, vector<rgw_buc
   return 0;
 }
 
+uint64_t MotrStore::get_new_req_id()
+{
+  uint64_t req_id = m0_dummy_id_generate();
```

Comment:
Maybe it would be better to use `ceph::util::generate_random_number<uint64_t>();` ? See also https://github.com/Seagate/cortx-rgw/pull/168.

Reply:
Good point. As it's mentioned in #168 , all reqs' ids were set to zero, and both this patch and #168  introduce id assignment. I would agree that using of existing RGW functions better.
The reason why motr function is used here comes from ADDB usage nature and its usage is intentional. `m0_dummy_id_generate()` function guarantees unique id for all callers within motr process. 
ADDB toolkit, PerfLine team developed, uses a tuple of (req_id, process_id) to distinguish each request in motr, regardless what layer request (rgw, client, dix, ioo, rpc, ...) is being processed. If request id source is not the same, then collisions may appear. 
For example, rgw req id is generated with `ceph::util::generate_random_number<uint64_t>()`  and gives `id=N`, and a dix request is generated with `m0_dummy_id_generate()` and it is also id=N. It means that during analysis time, collision of IDs is observed, and it spoils analysis picture.
There is a couple of options to handle it:
1. As RGW req_id may be assigned with an any number, just leave m0_dummy_id_generate(). It works fine and it is the simplest solution.
2. Use `ceph::util::generate_random_number<uint64_t>();` function and implement some more code to map RGW requests to RGW ADDB requests. A bit more code and a bit more complex.
Personally, I prefer to stick with the 1st option.

Reply:
What do you think, sir? @andriytk

Reply:
What's the probability of the collision?

Reply:
https://github.com/Seagate/cortx-rgw/pull/168 is merged already, so your code needs to be updated.

Reply:
I'd suggest avoiding "dummy" things. This function is not only used by ADDB, but also by S3 clients. If we use the "dummy" thing - it's very probably that all S3 clients will see relatively the same req ids in the responses - and this will be very inconvenient.

Also, I don't think the issue of collision is so critical for ADDB even when it happens (in a very unlucky case), so it can be ignored.

Reply:
What S3 clients are you talking about? Other s3 instances? - still you can distinguish them by FID, and get unique pair of (FID, request_id). 
As for collisions: you might have misunderstand my point - I don't even consider occasional collisions when there is a bullet-proof way of avoiding them.

Still "random" (or whatever you call it as an opposite of "dummy" thing) request id can be assigned, but MotrStore shall implement a map (lookup table) between "random" request_id and "addb" request id. As RGW is multi-threaded application, each thread takes request id from `RGWObjectCTX` parameter and the same request may be executed by different threads, so it's obvious why lookup is need. But how map is cleaned up? Is there a function call or method to indicate that a request processing is finished?